src/lib.rs

```diff
@@ -74,51 +75,115 @@ impl Notifier {
 unsafe impl Send for Notifier {}
 
 #[repr(C)]
-pub struct AzureCredentials {
-    account: *const c_char,
-    container: *const c_char,
-    key: *const c_char,
-    host: *const c_char,
+pub struct Config {
```

Comment:
tiny nit: maybe name this `RequestConfig` or something like that, to distinguish it more from the `GlobalConfigOptions`?

And/or maybe rename that something like `GlobalRuntimeConfig` or `GlobalDefaultsConfig`?

Reply:
I will leave this one with the generic `Config` name and rename the global one because the first deals not only with request related options but also credentials and client config.